koku/api/settings/utils.py

```diff
@@ -82,3 +88,54 @@ def create_dual_list_select(name, left_options=[], right_options=[], **kwargs):
     dual_list_select = {"component": "dual-list-select", "name": name}
     dual_list_select.update(**kwargs)
     return dual_list_select
+
+
+def create_select(name, **kwargs):
+    """
+    Create a select for the settings.
+
+    Args:
+        (String) name - unique name for switch.
+
+    Returns:
+        [Dict] - Subform component.
+    """
+    select = {"component": "select", "name": name}
+    select.update(**kwargs)
+    return select
+
+
+"""Common utilities and helpers for Currency."""
+
+
+def get_selected_currency_or_setup(schema):
+    with schema_context(schema):
+        if not CurrencySettings.objects.exists():
+            set_currency(schema)
+        currency = CurrencySettings.objects.all().first().currency
+        return currency
+
+
+def get_currency_options():
+    return [
+        dict(
+            value=currency.get("code"),
+            label=f"{currency.get('code')} ({currency.get('symbol')}) - {currency.get('name')}",
+        )
+        for currency in CURRENCIES
+    ]
+
+
+def set_currency(schema, currency_code=KOKU_DEFAULT_CURRENCY):
+    with schema_context(schema):
+        account_currency_setting = CurrencySettings.objects.all().first()
+        supported_currency_codes = [code.get("code") for code in CURRENCIES]
+
+        if currency_code not in supported_currency_codes:
+            raise ValueError(currency_code + " is not a supported currency")
```

Comment:
What is the user experience if the code encounters a bad currency code? For example, is an alert shown?

Reply:
No this would never be triggers by the UI at all. The only way this could ever be triggered is if you called the API directly with a currency that is not supported.